src/main/java/uk/gov/digital/ho/hocs/casework/api/CaseDataService.java

```diff
@@ -339,7 +341,31 @@ CaseSummary getCaseSummary(UUID caseUUID) {
         });
 
         return Stream.concat(auditTimeline, notesTimeline);
+    }
+
+    public Map<String, String> updateTeamByStageAndTexts(UUID caseUUID, UUID stageUUID, String stageType, String teamUUIDKey, String teamNameKey, String[] texts) {
+        log.debug("Updating Team by Stage: {} {}", stageUUID, stageType);
+        Map<String, String> dataMap = getCaseData(caseUUID).getDataMap(objectMapper);
+        // build the linkValue text string used to search the team link table by converting "text" key to the case's data value
+        String linkValue = null;
+        for (String text : texts){
+            String value = dataMap.getOrDefault(text, "");
+            if (!value.isEmpty()) {
+                if (linkValue != null) {
+                    linkValue += "_";
+                    linkValue += value;
+                } else {
+                    linkValue = value;
```

Comment:
Are we sure we want to reset the linkValue to blank string if the last text is not found? This would get rid of any previously allocated linkValues. I assume current logic (workflow) is built so that we always match on all text values, i'm not sure if this is always going to be the case going forward

Reply:
If the last text isn't found then `if (!value.isEmpty()) {` means nothing happens. Line 358 only happens when (i) the last text _is_ found and, (ii) linkValue is null (hasn't been loaded with anything yet, i.e. first time round).

Reply:
Yes, thats right...sorry i swear it looked different last time I looked at it :D